dockstore-integration-testing/src/test/java/io/dockstore/client/cli/SystemClientIT.java

```diff
@@ -363,28 +364,31 @@ public void testGetFiles() throws IOException, TimeoutException, ApiException {
                 .toolsIdVersionsVersionIdDockerfileGet("registry.hub.docker.com/seqware/seqware/test5", "master");
         assertTrue(toolDockerfile.getDockerfile().contains("dockerstuff"));
         final ToolDescriptor cwl = toolApi
-                .toolsIdVersionsVersionIdTypeDescriptorGet("cwl", "registry.hub.docker.com/seqware/seqware/test5", "master");
+                .toolsIdVersionsVersionIdTypeDescriptorGet("registry.hub.docker.com/seqware/seqware/test5", "master", "cwl");
         assertTrue(cwl.getDescriptor().contains("cwlstuff"));
 
         // hit up the plain text versions
         final String basePath = client.getBasePath();
         String encodedID = "registry.hub.docker.com%2Fseqware%2Fseqware%2Ftest5";
-        URL url = new URL(
-                basePath + DockstoreWebserviceApplication.GA4GH_API_PATH + "/tools/" + encodedID + "/versions/master/plain-CWL/descriptor");
+        URL url = UriBuilder.fromPath(basePath)
+                .path(DockstoreWebserviceApplication.GA4GH_API_PATH + "/tools/" + encodedID + "/versions/master/descriptor")
+                .queryParam("type", "PLAIN_CWL").build().toURL();
+
         List<String> strings = Resources.readLines(url, Charset.forName("UTF-8"));
         assertTrue(strings.size() == 1 && strings.get(0).equals("cwlstuff"));
 
         //hit up the relative path version
         String encodedPath = "%2FDockstore.cwl";
-        url = new URL(
-                basePath + DockstoreWebserviceApplication.GA4GH_API_PATH + "/tools/" + encodedID + "/versions/master/plain-CWL/descriptor/"
-                        + encodedPath);
+        url = UriBuilder.fromPath(basePath)
+                .path(DockstoreWebserviceApplication.GA4GH_API_PATH + "/tools/" + encodedID + "/versions/master/descriptor/" + encodedPath)
+                .queryParam("type", "PLAIN_CWL").build().toURL();
```

Comment:
Might want to create a new API path instead of re-using at `DockstoreWebserviceApplication.GA4GH_API_PATH`. 
And then deprecate the old one. For example, `/api/ga4gh/v1.1` instead of `/api/ga4gh/v1` 
This looks like a breaking change.